andela/travel_tool_front
# #162008362 Fix travel checklist submissions issues

#### What does this PR do?
This PR ensures that travel checklist submission process is seamless.

#### Description of Task to be completed?
- The travel checklist submission design should mirror the mock-up
- Refactor `ChecklistSubmissions APIs, sagas, reducers and actions`
- Refactor checklist submissions design to align with mock-up
- Refactor checklist submissions functionality
- Refactor tests to accommodate changes

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Enter into the project directory - `cd travel_tool_front`
- switch to this branch - `bg-fix-checklist-submissions-162008362`
- Install dependencies and start the  app - `make start`
-  [Start](https://github.com/andela/travel_tool_back/pull/174)  the backend.
- Navigate to [travela](http://travela-local.andela.com:3000) in your browser and login
- Go to the `DB` and on the `UserRoles` table, set yourself as a `Manager`
- Go to the browser, refresh the page,
- Make a `New Request` and approve it.
- On the approved request click the `ellipses`, a menu should drop down
- On the menu click on `Travel Checklist`
- A modal should pop up and you can go ahead to submit your checklist Item
- Close the modal and the `Travel Checklist column` should reflect the percentage completed of the checklist submission

#### Any background context you want to provide?
Ensure to fill the `.env` variables:
`REACT_APP_CLOUNDINARY_API=
REACT_APP_PRESET_NAME=
`
#### What are the relevant pivotal tracker stories?
[#162008362](https://www.pivotaltracker.com/story/show/162008362)

#### Screenshots (if appropriate)
![subm](https://user-images.githubusercontent.com/27569442/48838201-f913ea80-ed87-11e8-9ed5-a929ff6ab622.gif)


#### Questions:
N/A
